addons/base_import_module/models/ir_module.py

```diff
@@ -5,20 +6,20 @@
 from os.path import join as opj
 
 import openerp
-from openerp.osv import osv
+from openerp import _, models
 from openerp.tools import convert_file
-from openerp.tools.translate import _
 from openerp.exceptions import UserError
 
 _logger = logging.getLogger(__name__)
 
 MAX_FILE_SIZE = 100 * 1024 * 1024 # in megabytes
 
-class view(osv.osv):
+
+class View(models.Model):
     _inherit = "ir.module.module"
 
-    def import_module(self, cr, uid, module, path, force=False, context=None):
-        known_mods = self.browse(cr, uid, self.search(cr, uid, []))
+    def import_module(self, module, path, force=False):
+        known_mods = self.search([])
         known_mods_names = dict([(m.name, m) for m in known_mods])
```

Comment:
use a dict comprehension, or at least remove the useless intermediate list comprehension